src/__internal__/focus-trap/focus-trap.component.tsx

```diff
@@ -145,11 +146,29 @@ const FocusTrap = ({
       trapWrappers.forEach((ref) => {
         // istanbul ignore else
         if (ref.current) {
-          elements.push(
-            ...Array.from(ref.current.querySelectorAll(selector)).filter(
-              (el) => Number((el as HTMLElement).tabIndex) !== -1,
-            ),
+          const focusableElements = Array.from(
+            ref.current.querySelectorAll(selector),
+          ).filter((el) => Number((el as HTMLElement).tabIndex) !== -1);
+
+          elements.push(...focusableElements);
+
+          const scrollableElements = Array.from(
```

Comment:
comment: I don't think we need to have an array here as it will only ever return one or zero elements with how are modals are currently structured. We may also be able to use `contains` to check if the found element contains anything focusable

```
const scrollableElement = ref.querySelector(defaultScrollableSelectors);

if (!scrollableElement) {
  return;
}

if (
  scrollableElement.scrollHeight > scrollableElement.clientHeight && 
  (window.getComputedStyle(scrollableElement).overflowY === "scroll" || 
  window.getComputedStyle(scrollableElement).overflowY === "auto") &&
  !focusableElements.some(el => scrollableElement.contains(el))
) {
   scrollableElement.setAttribute("tabindex", "0");
}

Reply:
Happy for this to be non-blocking if you find any issues or prefer your way as as far as I can tell it works etc